src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -368,25 +386,38 @@ private void subscribe(String topic) {
         try {
             SubAckPacket subAckPacket = client.subscribe(subscribePacket)
                     .get(MQTT_OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
-            if (subAckPacket.getReasonCodes().stream().allMatch(this::subscriptionIsSuccessful)) {
-                synchronized (subscriptionsLock) {
-                    subscribedLocalMqttTopics.add(topic);
-                }
-                LOGGER.atDebug()
+            if (subAckPacket.getReasonCodes().stream().allMatch(this::retrySubscribe)) {
+                // received a retryable reason code
+                LOGGER.atError()
                         .kv(LOG_KEY_REASON_CODES, subAckPacket.getReasonCodes())
                         .kv(LOG_KEY_REASON, subAckPacket.getReasonString())
                         .kv(LOG_KEY_TOPIC, topic)
-                        .log("Successfully subscribed to topic");
+                        .log("Failed to subscribe to topic");
+                // throw exception to trigger RetryUtils
+                throw new RetryableMqttOperationException("Failed to subscribe to topic");
             } else {
+                if (subAckPacket.getReasonCodes().stream().allMatch(this::subscriptionIsSuccessful)) {
```

Comment:
nit: the ordering you had before was fine
```
if (subscription successful) {
   ...
} else if (retryable) {
   ...
} else {
   ...
}
```

Reply:
Ah I mistook your other comment about the structure, I'll revert it